src/cargo/core/compiler/mod.rs

```diff
@@ -1353,14 +1353,39 @@ fn check_cfg_args(build_runner: &BuildRunner<'_, '_>, unit: &Unit) -> Vec<OsStri
         // Cargo and docs.rs than rustc and docs.rs. In particular, all users of docs.rs use
         // Cargo, but not all users of rustc (like Rust-for-Linux) use docs.rs.
 
-        vec![
+        let mut args = vec![
             OsString::from("--check-cfg"),
             OsString::from("cfg(docsrs)"),
             OsString::from("--check-cfg"),
             arg_feature,
-        ]
+        ];
+
+        // Also include the custom arguments specified in `[lints.rust.unexpected_cfgs.check_cfg]`
+        if let Ok(Some(lints)) = unit.pkg.manifest().resolved_toml().resolved_lints() {
```

Comment:
We should remove the ["local only"](https://doc.rust-lang.org/nightly/cargo/reference/build-scripts.html#example-of-local-only-buildrsbuild-script) build script example as this is meant to target that use case?

Reply:
Yes, I think, I don't see any reason to use a "local only" build script over this new config. Removed.